cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -1108,47 +1126,18 @@ private ApplicationDetail enforceApplicationDetailAccess(ApplicationId appId,
   }
 
   // get filter for app specs by artifact name and version. if they are null, it means don't filter.
-  private Predicate<ApplicationDetail> getAppPredicate(Set<String> artifactNames, @Nullable String artifactVersion) {
+  private List<ApplicationFilter> getAppPredicates(Set<String> artifactNames, @Nullable String artifactVersion) {
     if (artifactNames.isEmpty() && artifactVersion == null) {
```

Comment:
handle the case when artifactNames is null ?

Reply:
no, null collection is usually a coding error

Reply:
then shall we update the comments above `if they are null, it means don't filter.` ?